Packs/Anomali_Enterprise/Integrations/Anomali_Enterprise/Anomali_Enterprise.py

```diff
@@ -0,0 +1,290 @@
+        # ae-authorization is needed only for get results
+        self._headers.update({'ae-authorization': f'{self._username}:{self._password}'})
+        params = {'jobid': job_id}
+        return self._http_request(method='GET', url_suffix='/api/v1/mars/forensic', headers=self._headers,
+                                  params=params)
+
+    def domain_request(self, domain: List[str]) -> dict:
+        """Retrieve information regarding a domain.
+        Args:
+            domain: the domain name to search
+        Returns:
+            Response from API.
+        """
+        data = json.dumps({'username': self._username, 'password': self._password, 'domains': domain})
+        return self._http_request(method='POST', url_suffix='/api/v1/mars/dga_score', headers=self._headers,
+                                  data=data)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def module(client: Client) -> str:
+    """
+    Performs basic get request
+    """
+    response = client.domain_request(argToList('google.com'))
+    if response.get('result') != 'success':
+        raise Exception('To Use Anomali Enterprise, make sure you are using the current username and password '
+                        'and have the needed permissions.')
+    return 'ok'
+
+
+def start_search_job(client: Client, args: dict) -> CommandResults:
+    """Start a search job for IOCs.
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+
+    Returns:
+        CommandResults.
+    """
+    from_ = str(args.get('from', '1 day'))
+    to_ = str(args.get('to', ''))
+    indicators = argToList(args.get('indicators'))
+
+    timestamp_format = '%Y-%m-%dT%H:%M:%S.%f'
+    from_iso = parse_date_range(from_, date_format=timestamp_format)[0]
+    if to_:
+        to_iso = parse_date_range(to_, date_format=timestamp_format)[0]
+    else:
+        to_iso = datetime.now().strftime(timestamp_format)
+
+    response = client.start_search_job_request(from_iso, to_iso, indicators)
+
+    start_search_outputs = {
+        'status': 'in progress',
+        'job_id': response.get('jobid', '')
+    }
+
+    return CommandResults(
+        outputs_prefix='AnomaliEnterprise.ForensicSearch',
+        outputs_key_field='job_id',
+        outputs=start_search_outputs,
+        readable_output=tableToMarkdown(name="Forensic search started:", t=start_search_outputs, removeNull=True),
+        raw_response=response
+    )
+
+
+def get_search_job_result(client: Client, args: Dict) -> CommandResults:
+    """Get the search job result.
+
+    Args:
+        client: Client object with request.
+        args: Usually demisto.args()
+
+    Returns:
+        CommandResults.
+    """
+    job_id = str(args.get('job_id'))
+
+    response = client.get_search_job_result_request(job_id)
+    if 'error' in response:
+        raise Exception(f"{str(response.get('error'))}. Job ID might have expired.")
+
+    response.update({'job_id': job_id})
+    if not response.get('complete'):
```

Comment:
Is it possible for a search to be cancelled or fail?

Reply:
yes it can fail. if so the `error` key will be in the response, I covered that in the code and in a UT.
did not find an indication that it can be cancelled.